src/game_api/entity.hpp

```diff
@@ -1,444 +1,249 @@
+    /// Triggers weapons and other held items like teleportter, mattock etc. You can check the [virtual-availability.md](https://github.com/spelunky-fyi/overlunky/blob/main/docs/virtual-availability.md), if entity has `open` in the `on_open` you can use this function, otherwise it does nothing. Returns false if action could not be performed (cooldown is not 0, no arrow loaded in etc. the animation could still be played thou)
+    virtual bool trigger_action(Entity* user) = 0;
+    /// Activates a button prompt (with the Use door/Buy button), e.g. buy shop item, activate drill, read sign, interact in camp, ... `get_entity(<udjat socket uid>):activate(players[1])` (make sure player 1 has the udjat eye though)
+    virtual void activate(Entity* activator) = 0;
+
+    virtual void on_collision2(Entity* other_entity) = 0; // needs investigating, difference between this and on_collision1, maybe this is on_hitbox_overlap as it works for logical tiggers
+    virtual uint16_t get_metadata() = 0;                  // e.g. for turkey: stores health, poison/curse state, for mattock: remaining swings (returned value is transferred)
+    virtual void apply_metadata(uint16_t metadata) = 0;
+    virtual void on_walked_on_by(Entity* walker) = 0;  // hits when monster/player walks on a floor, does something when walker.velocityy<-0.21 (falling onto) and walker.hitboxy * hitboxx > 0.09
+    virtual void on_walked_off_by(Entity* walker) = 0; // appears to be disabled in 1.23.3? hits when monster/player walks off a floor, it checks whether the walker has floor as overlay, and if so, removes walker from floor's items by calling virtual remove_item_ptr
+    virtual void on_ledge_grab(Entity* who) = 0;       // only ACTIVEFLOOR_FALLING_PLATFORM, does something with game menager
+    virtual void on_stood_on_by(Entity* entity) = 0;   // e.g. pots, skulls, pushblocks, ... standing on floors
+    virtual void toggle_backlayer_illumination() = 0;  // only for CHAR_*: when going to the backlayer, turns on player emitted light
+    virtual void v34() = 0;                            // only ITEM_TORCH, calls Torch.light_up(false), can't get it to trigger
+    virtual void liberate_from_shop() = 0;             // can also be seen as event: when you anger the shopkeeper, this function gets called for each item; can be called on shopitems individually as well and they become 'purchased'
+
+    /// Applies changes made in `entity.type`
+    virtual void apply_db() = 0; // This is actually just an initialize call that is happening once after  the entity is created
+};
+
+std::tuple<float, float, uint8_t> get_position(uint32_t uid);
+std::tuple<float, float, uint8_t> get_render_position(uint32_t uid);
+
+std::tuple<float, float> get_velocity(uint32_t uid);
+
+AABB get_hitbox(uint32_t uid, bool use_render_pos);
+
+Entity* get_entity_ptr(uint32_t uid);
+Entity* get_entity_ptr_local(uint32_t uid);
```

Comment:
github doesn't know how to chose the changes here, was this just moving out the entity db and other structs to a different file?

Reply:
Same as the .cpp